src/test/compile-fail/derived-errors/issue-31997.rs

```diff
@@ -20,7 +20,9 @@ fn closure<F, T>(x: F) -> Result<T, ()>
 }
 
 fn foo() -> Result<(), ()> {
-    try!(closure(|| bar(0 as *mut _))); //~ ERROR cannot find function `bar` in this scope
+    try!(closure(|| bar(0 as *mut _)));
+    //~^ ERROR cannot find function `bar` in this scope
+    //~^^ ERROR cannot cast to a pointer of an unknown kind
```

Comment:
So, clearly this isn't a *pure* refactoring -- at least the error messages changed. This new error doesn't appear particularly helpful -- I wonder if we should find a way to suppress it.

Reply:
The error suppression regressed here, don't know what's going on. An `is_tainted_by_errors()` check before emitting cast errors should fix this, not sure if it's a good fix.

Reply:
That seems like a reasonable thing to do, to me, or at least around cases involve "unknown" things like type variables.